### ISSUE TITLE
Zed logs error for missing ~/.config/github-copilot (linux)

### ISSUE THREAD

**Issue by anthonyryan1** [zed-industries/zed#17831]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

1. Update `settings.json` to disable GitHub Copilot, because I don't want the contents of my text editor sent to a 3rd party in the background.

```json
  "features": {
    "inline_completion_provider": "none"
  }
```

2. Startup Zed
3. Check Zed.log

```sh
grep -rni copilot ~/.local/share/zed/logs/Zed.log
```
```
2024-09-14T16:10:05.81434634-04:00 [ERROR] No such file or directory (os error 2) about ["/home/ar/.config/github-copilot"]
```

If copilot is properly disabled, why is it searching searching for it's config/API keys?

### Environment

Zed: v0.152.3 (Zed)
OS: Linux Wayland gentoo 2.15
Memory: 31.3 GiB
Architecture: x86_64
GPU: AMD Radeon RX 570 Series (RADV POLARIS10) || radv || Mesa 24.1.6

**Comment by sa1** [comment]
Apart from the inline provider, there's also the AI chat. I suspect Zed looks at which providers you're signed into for the chat.

**Comment by vuldin** [comment]
One result of this issue is that Zed fails to startup (just hangs during startup):
```
❯ zed --foreground
[2024-09-16T10:09:16-05:00 ERROR fs] No such file or directory (os error 2) about ["/home/vuldin/.config/github-copilot"]
[2024-09-16T10:09:17-05:00 ERROR assistant] no worktrees when constructing ProjectLspAdapterDelegate
```

I must then kill the process with `pkill zed`.

**Comment by anthonyryan1** [comment]
Not currently preventing startup for me. No current lead on why that part differs for you.

**Comment by notpeter** [comment]
Yeah, looking at the code the GitHub Copilot Chat via the Assistant panel uses that file:

https://github.com/zed-industries/zed/blob/0f4ebdfbca721614f3cadafc3b44e4fbf099afda/crates/copilot/src/copilot_chat.rs#L175-L187

I was able to reproduce by removing my `.config/github-copilot/` directory and restarting Zed.  I tried on Linux/MacOS, but the error only showed on linux.

**Comment by notpeter** [comment]
I did some digging and I believe this is happening because we add a watch for `~/.config/github-copilot/hosts.json`.
https://github.com/zed-industries/zed/blob/f5124c21d1f6a5e52e1d853229b50021d8b74eb9/crates/copilot/src/copilot_chat.rs#L195-L199

If the `github-copilot` directory does not exist, then attempting to add a watch for a file in a directory that does not exist will generate an error.

**Comment by notpeter** [comment]
I believe this was fixed by:
- https://github.com/zed-industries/zed/pull/21360

More info [here](https://github.com/zed-industries/zed/issues/21255#issuecomment-2508940898)

If you are still seeing this in your logs with a current release, please comment with more details and I'm happy to reopen.
